src/plugins/GUITestBase/src/GTUtilsDashboard.cpp

```diff
@@ -146,6 +175,80 @@ bool GTUtilsDashboard::hasNotifications(HI::GUITestOpStatus &os) {
     return notificationsWidget->isVisible();
 }
 
+#define GT_METHOD_NAME "getNotificationTypeFromHtml"
+GTUtilsDashboard::Notification::Type GTUtilsDashboard::getNotificationTypeFromHtml(HI::GUITestOpStatus &os,
+    const QString &html) {
+    using Notification = U2::GTUtilsDashboard::Notification;
+    const QString errMsg = "Dashboard notification type not found";
+    GT_CHECK_RESULT(!html.isEmpty(), errMsg, Notification::Type::ERR)
+
+    QXmlStreamReader reader(html);
```

Comment:
I see big overdesign and non-needed complexity. Instead of using SAX parser (or DOM model) just locate the info type in the string like: html.indexOf(""<center><img class=\""")...
As the result this method may be 1-2 lines long.